static/js/metrics-explorer.js

```diff
@@ -1965,91 +1965,50 @@ const shouldShowLegend = (panelId, datasets) => {
 async function convertDataForChart(data) {
     let seriesArray = [];
 
-    if (Object.prototype.hasOwnProperty.call(data, 'series') && Object.prototype.hasOwnProperty.call(data, 'timestamps') && Object.prototype.hasOwnProperty.call(data, 'values')) {
+    if (data.series && data.timestamps && data.values) {
         let chartStartTime = data.startTime;
         let chartEndTime = Math.floor(Date.now() / 1000);
         const timeRange = chartEndTime - chartStartTime;
-        // // Determine the best time unit based on the time range
-        if (timeRange > 365 * 24 * 60 * 60) {
-            timeUnit = 'month';
-        } else if (timeRange >= 90 * 24 * 60 * 60) {
-            timeUnit = '7day';
-        } else if (timeRange >= 30 * 24 * 60 * 60) {
-            timeUnit = '2day';
-        } else if (timeRange >= 7 * 24 * 60 * 60) {
-            timeUnit = '12hour';
-        } else if (timeRange >= 2 * 24 * 60 * 60) {
-            timeUnit = '6hour';
-        } else if (timeRange >= 24 * 60 * 60) {
-            timeUnit = '3hour';
-        } else if (timeRange >= 12 * 60 * 60) {
-            timeUnit = '30minute';
-        } else if (timeRange >= 3 * 60 * 60) {
-            timeUnit = '15minute';
-        } else if (timeRange >= 30 * 60) {
-            timeUnit = '5minute';
-        } else {
-            timeUnit = 'minute';
-        }
+
+        // Determine the best time unit based on the time range
+        timeUnit = determineTimeUnit(timeRange);
+
+        let calculatedInterval = data.intervalSec;
+
         for (let i = 0; i < data.series.length; i++) {
             let series = {
                 seriesName: data.series[i],
                 values: {},
             };
-            //eslint-disable-next-line no-useless-escape
```

Comment:
why was this code removed?

Reply:
The switch statement was causing inconsistencies in plotting the values, so I simplified the logic. Also, the regex wasn't correctly identifying if it was a numeric expression or a query.